File: app/schemas/consumption.py
from pydantic import BaseModel, Field, ConfigDict
from decimal import Decimal
from datetime import datetime


class ClientConsumptionCreate(BaseModel):
    stay_id: int = Field(..., gt=0)
    product_id: int = Field(..., gt=0)
    quantity: int = Field(..., gt=0)
    notes: str | None = Field(default=None, max_length=5000)

class ClientConsumptionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    hotel_id: int
    stay_id: int
    product_id: int
    quantity: int
    unit_price: Decimal
    created_at: datetime
    is_cancelled: bool
    cancelled_at: datetime | None
    cancelled_by: int | None

File: frontend/src/types/consumption.ts
export interface ConsumptionCreate {
    stay_id: number
    product_id: number
    quantity: number
    notes: string | null
}

export interface ConsumptionRead {
    id: number
    hotel_id: number
    stay_id: number
    product_id: number
    quantity: number
    unit_price: number
    created_at: string
    is_cancelled: boolean
    cancelled_at: string | null
    cancelled_by: number | null
}